test_launcher fails on the ARM64 Windows Non-Debug Azure 3.x buildbot

test_launcher fails on the ARM64 Windows Non-Debug Azure 3.x buildbot:
https://buildbot.python.org/all/#/builders/897/builds/322

The failure may be caused by https://github.com/python/cpython/commit/73473fdeac3ff9d75ac9d189bb8552b7459812f1

```
0:14:26 Re-running test_launcher in verbose mode (matching: test_virtualenv_with_env)
test_virtualenv_with_env (test.test_launcher.TestLauncher.test_virtualenv_with_env) ... FAIL
Stdout:
++ COMMAND ++
[WindowsPath('C:/Workspace/buildarea/3.x.linaro2-win-arm64.nondebug/build/PCbuild/arm64/py.exe'), '-3']
++ STDOUT ++
++ STDERR ++
argv0: C:\Workspace\buildarea\3.x.linaro2-win-arm64.nondebug\build\PCbuild\arm64\py.exe
version: 3.12.0a0
# Reading from C:\Users\tcwg\AppData\Local\py.ini for defaults/py_python3
# Did not find file C:\Users\tcwg\AppData\Local\py.ini
# Reading from C:\Workspace\buildarea\3.x.linaro2-win-arm64.nondebug\build\PCbuild\arm64\py.ini for defaults/py_python3
# Did not find file C:\Workspace\buildarea\3.x.linaro2-win-arm64.nondebug\build\PCbuild\arm64\py.ini
SearchInfo.originalCmdLine: C:\Workspace\buildarea\3.x.linaro2-win-arm64.nondebug\build\PCbuild\arm64\py.exe -3
SearchInfo.restOfCmdLine: 
SearchInfo.executablePath: (null)
SearchInfo.scriptFile: (null)
SearchInfo.executable: python.exe
SearchInfo.executableArgs: (null)
SearchInfo.company: (null)
SearchInfo.tag: 3
SearchInfo.oldStyleTag: True
SearchInfo.lowPriorityTag: False
SearchInfo.exclude32Bit: False
SearchInfo.only32Bit: False
SearchInfo.allowDefaults: True
SearchInfo.allowExecutableOverride: True
SearchInfo.windowed: False
SearchInfo.list: False
SearchInfo.listPaths: False
SearchInfo.help: False
# PythonTestSuite/ignored has no executable path
 -V:ContinuumAnalytics/Anaconda39-64 C:\wenv\x64_vs2022\conda\python.exe
 -V:PythonTestSuite/3.100 X.Y.exe
 -V:PythonTestSuite/3.100-32 X.Y-32.exe
 -V:PythonTestSuite/3.100-arm64 X.Y-arm64.exe -X fake_arg_for_test
No suitable Python runtime found
Pass --list (-0) to see all detected environments on your machine
or set environment variable PYLAUNCHER_ALLOW_INSTALL to use winget
or open the Microsoft Store to the requested version.
======================================================================
FAIL: test_virtualenv_with_env (test.test_launcher.TestLauncher.test_virtualenv_with_env)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Workspace\buildarea\3.x.linaro2-win-arm64.nondebug\build\Lib\test\test_launcher.py", line 487, in test_virtualenv_with_env
    data2 = self.run_py(["-3"], env={**env, "PY_PYTHON": "-3"})
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Workspace\buildarea\3.x.linaro2-win-arm64.nondebug\build\Lib\test\test_launcher.py", line 223, in run_py
    self.assertEqual(expect_returncode, p.returncode)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 0 != 103
Stdout:
++ COMMAND ++
[WindowsPath('C:/Workspace/buildarea/3.x.linaro2-win-arm64.nondebug/build/PCbuild/arm64/py.exe'), '-3']
++ STDOUT ++
++ STDERR ++
argv0: C:\Workspace\buildarea\3.x.linaro2-win-arm64.nondebug\build\PCbuild\arm64\py.exe
version: 3.12.0a0
# Reading from C:\Users\tcwg\AppData\Local\py.ini for defaults/py_python3
# Did not find file C:\Users\tcwg\AppData\Local\py.ini
# Reading from C:\Workspace\buildarea\3.x.linaro2-win-arm64.nondebug\build\PCbuild\arm64\py.ini for defaults/py_python3
# Did not find file C:\Workspace\buildarea\3.x.linaro2-win-arm64.nondebug\build\PCbuild\arm64\py.ini
SearchInfo.originalCmdLine: C:\Workspace\buildarea\3.x.linaro2-win-arm64.nondebug\build\PCbuild\arm64\py.exe -3
SearchInfo.restOfCmdLine: 
SearchInfo.executablePath: (null)
SearchInfo.scriptFile: (null)
SearchInfo.executable: python.exe
SearchInfo.executableArgs: (null)
SearchInfo.company: (null)
SearchInfo.tag: 3
SearchInfo.oldStyleTag: True
SearchInfo.lowPriorityTag: False
SearchInfo.exclude32Bit: False
SearchInfo.only32Bit: False
SearchInfo.allowDefaults: True
SearchInfo.allowExecutableOverride: True
SearchInfo.windowed: False
SearchInfo.list: False
SearchInfo.listPaths: False
SearchInfo.help: False
# PythonTestSuite/ignored has no executable path
 -V:ContinuumAnalytics/Anaconda39-64 C:\wenv\x64_vs2022\conda\python.exe
 -V:PythonTestSuite/3.100 X.Y.exe
 -V:PythonTestSuite/3.100-32 X.Y-32.exe
 -V:PythonTestSuite/3.100-arm64 X.Y-arm64.exe -X fake_arg_for_test
No suitable Python runtime found
Pass --list (-0) to see all detected environments on your machine
or set environment variable PYLAUNCHER_ALLOW_INSTALL to use winget
or open the Microsoft Store to the requested version.
```

--- a/Lib/test/test_launcher.py
+++ b/Lib/test/test_launcher.py
@@ -245,7 +245,7 @@ class RunPyMixin:
             file.unlink()
 
     @contextlib.contextmanager
-    def test_venv(self):
+    def fake_venv(self):
         venv = Path.cwd() / "Scripts"
         venv.mkdir(exist_ok=True, parents=True)
         venv_exe = (venv / Path(sys.executable).name)
@@ -462,7 +462,7 @@ class TestLauncher(unittest.TestCase, RunPyMixin):
         self.assertEqual("PythonTestSuite/3.100", default)
 
     def test_virtualenv_in_list(self):
-        with self.test_venv() as (venv_exe, env):
+        with self.fake_venv() as (venv_exe, env):
             data = self.run_py(["-0p"], env=env)
             for line in data["stdout"].splitlines():
                 m = re.match(r"\s*\*\s+(.+)$", line)
@@ -482,9 +482,9 @@ class TestLauncher(unittest.TestCase, RunPyMixin):
                 self.fail("did not find active venv entry")
 
     def test_virtualenv_with_env(self):
-        with self.test_venv() as (venv_exe, env):
-            data1 = self.run_py([], env={**env, "PY_PYTHON": "-3"})
-            data2 = self.run_py(["-3"], env={**env, "PY_PYTHON": "-3"})
+        with self.fake_venv() as (venv_exe, env):
+            data1 = self.run_py([], env={**env, "PY_PYTHON": "PythonTestSuite/3"})
+            data2 = self.run_py(["-3"], env={**env, "PY_PYTHON": "PythonTestSuite/3"})
         # Compare stdout, because stderr goes via ascii
         self.assertEqual(data1["stdout"].strip(), str(venv_exe))
         self.assertEqual(data1["SearchInfo.lowPriorityTag"], "True")

--- a/PC/launcher2.c
+++ b/PC/launcher2.c
@@ -972,9 +972,6 @@ checkDefaults(SearchInfo *search)
         if (!slash) {
             search->tag = tag;
             search->tagLength = n;
-            // gh-92817: allow a high priority env to be selected even if it
-            // doesn't match the tag
-            search->lowPriorityTag = true;
         } else {
             search->company = tag;
             search->companyLength = (int)(slash - tag);
@@ -982,6 +979,9 @@ checkDefaults(SearchInfo *search)
             search->tagLength = n - (search->companyLength + 1);
             search->oldStyleTag = false;
         }
+        // gh-92817: allow a high priority env to be selected even if it
+        // doesn't match the tag
+        search->lowPriorityTag = true;
     }
 
     return 0;

